// DatePicker/DatePicker.js
import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Calendar from '../Calendar/Calendar'
import { clone } from '../dateHelpers'
import { datePickerStyles } from './styles'

const StyledDatePicker = styled.div`${datePickerStyles}`

class DatePicker extends React.Component {
  constructor(props) {
    super(props)
    const def = props.selected || new Date()
    this.state = {
      view: clone(def),
      selected: clone(def),
      minDate: null,
      maxDate: null,
    }
  }

  setMinDate = date => {
    this.setState({ minDate: date })
  }

  setMaxDate = date => {
    this.setState({ maxDate: date })
  }

  onChange = day => {
    this.setState({ selected: day })

    if (this.props.onChange) {
      this.props.onChange(day)
    }
  }

  render() {
    const {
      view,
      selected,
      minDate,
      maxDate,
    } = this.state

    const {
      children,
      ...rest
    } = this.props

    return (
      <StyledDatePicker {...rest}>
        <Calendar
          view={view}
          selected={selected}
          onSelect={this.onChange}
          minDate={minDate}
          maxDate={maxDate}
          {...rest}
        />
        { children }
      </StyledDatePicker>
    )
  }
}

DatePicker.propTypes = {
  selected: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
  activeColor: PropTypes.string,
  disabledColor: PropTypes.string,
  otherMonthColor: PropTypes.string,
  borderRadius: PropTypes.string,
  borderColor: PropTypes.string,
  ArrowLeft: PropTypes.string,
  ArrowRight: PropTypes.string,
  MonthName: PropTypes.string,
  Day:PropTypes.string,
  DayOfWeek: PropTypes.string,
}

DatePicker.defaultProps = {
  ArrowLeft: () => (<span>prev</span>),
  ArrowRight: () => (<span>next</span>),
  MonthName: ({ children }) => (<span>{children}</span>),
  Day: ({ children }) => (<span>{children}</span>),
  DayOfWeek: ({ children }) => (<span>{children}</span>),
  activeColor: 'red',
  disabledColor: 'green',
  otherMonthColor: 'green',
  borderRadius: '2px',
  borderColor: '#f7f7f7f',
}

export default DatePicker
