// components/DateUtils/DateUtils.js
const pad = (value, length) => {
  while (value.length < length)
    value = "0" + value;
  return value
}

const clone = date => {
  return new Date(date.getFullYear(), date.getMonth(), date.getDate(), date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds());
}

const toString = date => {
  return date.getFullYear() + "-" + pad((date.getMonth()+1).toString(), 2) + "-" + pad(date.getDate().toString(), 2);
}

const toDayOfMonthString = date => {
  return pad(date.getDate().toString())
}

const toMonthAndYearString = date => {
  var months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
  return months[date.getMonth()] + " " + date.getFullYear();
}

const moveToDayOfWeek = (date, dayOfWeek) => {
  while (date.getDay() !== dayOfWeek)
    date.setDate(date.getDate()-1)
  return date
}

const isSameDay = (first, second) => {
  return first.getFullYear() === second.getFullYear() && first.getMonth() === second.getMonth() && first.getDate() === second.getDate()
}

const isBefore = (first, second) => {
  return first.getTime() < second.getTime()
}

const isAfter = (first, second) => {
  return first.getTime() > second.getTime()
}

export {
  pad,
  clone,
  toString,
  toDayOfMonthString,
  toMonthAndYearString,
  moveToDayOfWeek,
  isSameDay,
  isBefore,
  isAfter,
}
