// components/DatePicker/DatePicker.js
import React from 'react'
import styled from 'styled-components'
import Calendar from '../Calendar'
import { clone } from '../DateUtils'
import { datePickerStyles } from './styles'

const StyledDatePicker = styled.div`${datePickerStyles}`

class DatePicker extends React.Component {
  constructor(props) {
    super(props)
    const def = props.selected || new Date()
    this.state = {
      view: clone(def),
      selected: clone(def),
      minDate: null,
      maxDate: null,
    }
  }

  setMinDate = date => {
    this.setState({ minDate: date })
  }

  setMaxDate = date => {
    this.setState({ maxDate: date })
  }

  onSelect = day => {
    this.setState({ selected: day })

    if (this.props.onSelect) {
      this.props.onSelect(day)
    }
  }

  render() {
    const {
      view,
      selected,
      minDate,
      maxDate,
    } = this.state

    const {
      children,
      ...rest
    } = this.props

    return (
      <StyledDatePicker {...rest}>
        <Calendar
          view={view}
          selected={selected}
          onSelect={this.onSelect}
          minDate={minDate}
          maxDate={maxDate}
        />
        { children }
      </StyledDatePicker>
    )
  }
}

export default DatePicker
