// Week/styles.js
import { css } from 'styled-components'

const variantStyles = ({
  otherMonth,
  disabled,
  selected,
  activeColor,
  otherMonthColor,
  monthBackgroundColor,
  disabledColor,
}) => {
  if (selected) {
    return css`
      color: ${activeColor};
      background: ${monthBackgroundColor};
      &:before {
        opacity: 1;
      }
      `
  }
  if (otherMonth) {
    return css`
      color: ${otherMonthColor};
    `
  }
  if (disabled) {
    return css`
      color: ${disabledColor};
    `
  }
  return css`
    background: ${monthBackgroundColor};
    &:hover:before {
      opacity: 0.25;
    }
  `
}

const dayStyles = ({
  ...rest
}) => css`
  position: relative;
  width: ${100/7}%;
  padding: 2% 0;
  cursor: pointer;
  transition: opacity 200ms;

  &:before {
    transition: opacity 200ms;
    opacity: 0;
    content: '';
    padding-top: 100%;
    width: 100%;
    position: absolute;
    border: 1px solid;
    border-radius: 50%;
    background: white;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%) scale(0.4);
    min-height: 75px;
    min-width: 75px;
  }

  * {
    color: inherit !important;
    position: relative;
    z-index: 1;
  }

  ${variantStyles(rest)}

`
const weekStyles = css`
  display: flex;
  flex-wrap: wrap;
`

export {
  weekStyles,
  dayStyles,
}
